src/main/scala/Util.scala

```diff
@@ -78,3 +78,33 @@ class PutSeqDriver(val s: Seq[Tuple2[BigInt,Int]])(implicit p: Parameters) exten
 
 }
 
+// Use gray coding to safely synchronize a word across a clock crossing.
+// This should be placed in the receiver's clock domain.
+class WordSync[T <: Data](gen: T, lat: Int = 2) extends Module {
+  val size = gen.cloneType.fromBits(UInt(0)).asUInt().getWidth
+  val io = new Bundle {
+    val in = gen.cloneType.flip
+    val out = gen.cloneType
+    val tx_clock = Clock(INPUT)
+  }
+  val bin2gray = Module(new BinToGray(gen,io.tx_clock))
+  val out_gray = ShiftRegister(bin2gray.io.gray, lat)
+  io.out := gen.cloneType.fromBits((0 until size).map{ out_gray.asUInt >> UInt(_) }.reduceLeft(_^_))
```

Comment:
The unneeded XORs will just get optimized away by the synthesis tool, no?

I'm not averse to bumping Chisel, though.